app/javascript/react/components/ReviewForm.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react'
+import _ from 'lodash'
+import { Redirect } from 'react-router'
+import ErrorList from './ErrorList'
+
+const ReviewForm = props => {
+  const [shouldRedirect, setShouldRedirect] = useState(false)
+  const [formData, setFormData] = useState("")
+  const {id} = props.match.params
+
+  const handleChange = (event) => {
+    setFormData(event.currentTarget.value)
+  }
+
+  const handleSubmit = event => {
+    event.preventDefault()
+    if (validateSubmission()){
+      postNewReview()
+    }
+  }
+
+  const [errors, setErrors] = useState([])
+  
+  const validateSubmission = () => {
+    let submitErrors = []
+    const requiredField = ["description"]
+    if (formData.trim() === ""){
+     submitErrors.push("Description can't be blank") 
+    }
+    if (formData.length < 20){
+      submitErrors.push("Description is too short")
```

Comment:
We can apply these validations here on the front end client side, but we should still have error handling as well coming from the database level. Not a required change, but something to consider - I would recommend validating more for presence here on the client side (checking the data after `trim`), and more specific validations to be handled by the back end database